### PR TITLE
Fix deploy error and deploy test case

### LINKED ISSUE #2333: markbind deploy error after migrating Site/index.js to TypeScript
### Please confirm that you have searched existing issues in the repo

Yes, I have searched the existing issues

### Any related issues?

_No response_

### Tell us about your environment

Windows 10

### MarkBind version

Master branch

### Describe the bug and the steps to reproduce it

Issue discovered after V5.0.0 is released. See detailed info and what has been done about it in this follow-up PR: https://github.com/MarkBind/markbind/pull/2331

TLDR:
- users cannot call markbind deploy while other functionalities seems alright
- by reverting the Site/index.ts back to Site/index.js (plus required changes), the issue seems fixed
- we still want to get to the root of things, namely:
  - review the original migration PR: https://github.com/MarkBind/markbind/pull/2270
  - find out which part causes the markbind deploy issue
  - re-migrate index.js back to index.ts, ensuring all changes have been applied properly
  - possibly add a test case to ensure markbind deploy is properly tested

### Expected behavior

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by WillCWX**
**What is the purpose of this pull request?**

- [ ] Documentation update
- [X] Bug fix
- [ ] Feature addition or enhancement
- [ ] Code maintenance
- [ ] DevOps
- [ ] Improve developer experience
- [ ] Others, please explain:

Close #2333
Related to #2331

**Overview of changes:**

- Migrate `Site/index.js` to `TypeScript` again.
- Change `options.remote` to be `defaultOptionConfig.remote` instead of `''`
  - This fixes the deploy error
- Remove `options.remote = defaultDeployConfig.remote` in `getDepUrl`
  - This fixes the false negative for the deploy test case 

**Anything you'd like to highlight/discuss:**

No

**Testing instructions:**

set up a test repo that with github pages 
run markbind deploy

**Proposed commit message: (wrap lines at 72 characters)**

Fix deploy error and deploy test case
---

**Checklist:** :ballot_box_with_check:

- [ ] Updated the documentation for feature additions and enhancements
- [ ] Added tests for bug fixes or features
- [X] Linked all related issues
- [X] No unrelated changes

### CODE REVIEW

File: packages/core/template/default/index.md
Comment (tlylt) [inline packages/core/template/default/index.md:1]: empty? same for a few more

Comment (WillCWX) [reply packages/core/template/default/index.md:1]: I have no idea why it writes empty when the file is still there...

Comment (WillCWX) [reply packages/core/template/default/index.md:1]: Apparently it is a windows vs linux / unix file thing: https://stackoverflow.com/questions/1257592/how-do-i-remove-files-saying-old-mode-100755-new-mode-100644-from-unstaged-cha

I'll see if I can remove that change from the commits

Comment (WillCWX) [reply packages/core/template/default/index.md:1]: I fixed the file permissions back to 100755

Comment (tlylt) [reply packages/core/template/default/index.md:1]: so what's the diff in this file? can we exclude it??

File: packages/core/template/default/site.json [hunk 1]
```diff
@@ -25,4 +25,4 @@
   "deploy": {
     "message": "Site Update."
   }
-}
```
Comment (tlylt) [inline packages/core/template/default/site.json:28]: same here can we remove this diff

File: packages/core/src/Site/index.ts [hunk 2]
```diff
@@ -584,7 +673,7 @@ class Site {
    * if there is a change in the variables file
```
Comment (tlylt) [inline packages/core/src/Site/index.ts:673]: image

**Review by tlylt — CHANGES_REQUESTED** [review]
Thank you @WillCWX for working on this PR. I have a few comments. Main issues:
- Could you take another look at the original PR and the changes made? https://github.com/MarkBind/markbind/pull/2270/files
  - I'm not sure why some changes (deletion, type assertions) that were made are not transferred back in this PR. I have pointed out a few places but I realized they are not done properly. If you have any concerns or comments as to why those are not necessary changes, please indicate for discussion.
- After everything is done, will need to re-organize into 2 commits as specified in the DG, one for rename, and one for migration. (Please rebase first as well)

File: packages/core/src/Site/index.ts [hunk 3]
```diff
@@ -160,11 +244,11 @@ class Site {
    * Util Methods
    */
 
-  static async rejectHandler(error, removeFolders) {
+  static async rejectHandler(error: unknown, removeFolders: string[]) {
     logger.warn(error);
     try {
       await Promise.all(removeFolders.map(folder => fs.remove(folder)));
-    } catch (err) {
+    } catch (err: any) {
```
[the remark this answers is not in the document]
Comment (WillCWX) [reply packages/core/src/Site/index.ts:251]: There is no real issue here, just a stylistic choice.
I will change it back

**Comment by WillCWX** [comment]
I've compared the current migration with the previous PR and also the v5.0.0 file and made the changes accordingly.
I have followed the previous versions, except for some small comment (deleting extra spaces) / positioning changes.

As requested, I've removed the comments noted and also all the `{Type}` in `@param {Type}`.

**Comment by WillCWX** [comment]
If its ok, I'll merge with master to update the branch and then rebase into two commits as mentioned in the docs.

**Comment by tlylt** [comment]
sure

**Comment by tlylt** [comment]
👍 @WillCWX can organize and will merge after that's done

**Review by tlylt — APPROVED** [review]

**Pull request merged**